src/clients/HubPoolClient.ts

```diff
@@ -703,7 +776,12 @@ export class HubPoolClient extends BaseAbstractClient {
       if (isUBA(version)) {
         // runningBalances array is a concatenation of pre-UBA runningBalances and incentiveBalances.
         executedRootBundle.runningBalances = runningBalances.slice(0, nTokens);
-        executedRootBundle.incentiveBalances = runningBalances.slice(nTokens);
+        // If bundle hasn't added incentive balance values, then assume they are 0. This is in place
+        // to make sure that the incentiveBalances isn't undefined.
+        executedRootBundle.incentiveBalances =
+          runningBalances.length > nTokens
+            ? runningBalances.slice(nTokens)
+            : Array(nTokens).fill(ethers.constants.Zero);
```

Comment:
This loosens the validation logic a lot and opens up for non-conformant runningBalances. The UMIP is very specific about runningBalances being 2x the length of l1Tokens, and I like that because it's easy to validate. Is it really necessary to change it?

![2023-07-19-152131_1017x300_scrot](https://github.com/across-protocol/sdk-v2/assets/108695806/2772f585-5b53-43ce-b9dc-b58d9ef82006)

Reply:
so I added this so that it can still validate pre UBA bundles. I'll remove it for now but I think we might run into problems on prod without this

Reply:
Actually I think this can all be simpler. Its not correct as is because we need to compare the bundle start block, not the proposal block, against the config store version. I think its easier to just check here if running balances is 2x l1 tokens or 1x. We don't need to check version https://github.com/across-protocol/sdk-v2/commit/97d0b647fcdb51e82900ccdc79c0330dae318c77

Reply:
> Actually I think this can all be simpler. Its not correct as is because we need to compare the bundle start block, not the proposal block, against the config store version. I think its easier to just check here if running balances is 2x l1 tokens or 1x. We don't need to check version [97d0b64](https://github.com/across-protocol/sdk-v2/commit/97d0b647fcdb51e82900ccdc79c0330dae318c77)

Just to be sure on that, we previously agreed that it was important to verify that the format complies with the version (I agree with this):
https://github.com/across-protocol/sdk-v2/pull/213#discussion_r1197043597

On the applicable block number to determine the version, we also discussed this previously and I'm pretty confident that it will work correctly (forwards and backwards) by evaluating on the proposal block number. There are potentially some TBDs on this though - i.e. it's very much dependent on how the UBA migration is handled.
https://github.com/across-protocol/sdk-v2/pull/230#discussion_r1223131959

Reply:
what if the UBA VERSION is updated after the first UBA bundle start block but before the proposal block number?

Checking the version at the proposal block  would return 2

Checking the version at the bundle start block would return 1

The correct version for this bundle should be 1 right?

Reply:
> The correct version for this bundle should be 1 right?

Yeah, but that presupposes a specific upgrade path. My assumption was that we'd need to pause deposits, fill all outstanding deposits and then propose a bundle before bumping the version. If we did that and then re-enabled deposits then we should have a clean upgrade where the same version applies to both the opening block and the proposal block. It does however have the downside of being open to a dispute if the upgrade proceeds within the challenge window, though. So if we can get over the version hump without all that overhead then that's great :+1: 

The removal of the version guard still strikes me as a step backwards - I'd prefer it if we were strict about the length as a function of the applicable version (i.e. I think your feedback on this in May was definitely valid).

Reply:
Update - I've been convinced the other way out of band :+1: It's probably not necessary to validate the length based on the version number because the bundle has already cleared liveness, meaning that consensus is that it was correct.